ctapipe/tools/train_particle_classifier.py

```diff
@@ -193,8 +207,26 @@ def _read_input_data(self, tel_type):
             feature_names=feature_names,
             rng=self.rng,
             log=self.log,
-            n_events=self.n_background.tel[tel_type],
+            n_events=n_background,
         )
+        if n_events is None:  # use as many events as possible (keeping signal_fraction)
+            if len(signal) < len(background) / (1 / self.signal_fraction - 1):
```

Comment:
I think using the condition written in the form `N_tot*signal_fraction > N_signal` is much clearer than the formulation you are using here, so 
```
len(signal) < (len(signal) + len(background))*self.signal_fraction 
```
but preferably you define the constants `N_s` and `N_b` as the lengths of the signal and background input samples, and use those instead of `len(signal)` and `len(background)` in the above expression.